packages/promotion/src/types/promotion-rule.ts

```diff
@@ -0,0 +1,11 @@
+import { PromotionRuleOperator } from "@medusajs/types"
+
+export interface CreatePromotionRuleDTO {
```

Comment:
question: This type lives in both `types` and `promotion`, but is different. Is that intentional?

Reply:
Yes, the service layer types are (potentially) different from the entry point service types. This is a convention we adopted even though there can be a great level of commonality or no change at all bw these 2. Among some other benefits that I can't seem to recall atm, it should make it easier to extend the entry point services without affecting the internal services.

Reply:
I confirm, that they have different responsibilities and can evolve differently, what is expected by the entry point is not necessarily what is expected by the service layer, it can be altered by the module service. Even though as mentioned by Riqwan it can look alike, they can evolve very differently as they are different layers.

Reply:
Got it, thanks for explaining 👍